core/src/main/java/io/questdb/griffin/engine/functions/groupby/LastArrayGroupByFunction.java

```diff
@@ -0,0 +1,156 @@
+    private final Function arg;
+    private final GroupByArraySink sink;
+    private int valueIndex;
+
+    public LastArrayGroupByFunction(@NotNull Function arg) {
+        this.arg = arg;
+        this.type = arg.getType();
+        this.sink = new GroupByArraySink(type);
+    }
+
+    @Override
+    public void clear() {
+        sink.of(0);
+    }
+
+    @Override
+    public void computeFirst(MapValue mapValue, Record record, long rowId) {
+        mapValue.putLong(valueIndex, rowId);
+        sink.of(0);
+        sink.put(arg.getArray(record));
+        mapValue.putLong(valueIndex + 1, sink.ptr());
+    }
+
+    @Override
+    public void computeNext(MapValue mapValue, Record record, long rowId) {
+        mapValue.putLong(valueIndex, rowId);
+        long ptr = mapValue.getLong(valueIndex + 1);
+        sink.of(ptr);
+        sink.put(arg.getArray(record));
+        mapValue.putLong(valueIndex + 1, sink.ptr());
+    }
+
+    @Override
+    public Function getArg() {
+        return this.arg;
+    }
+
+    @Override
+    public ArrayView getArray(Record rec) {
+        long ptr = rec.getLong(valueIndex + 1);
+        sink.of(ptr);
+        ArrayView array = sink.getArray();
+        if (array == null) {
+            return ArrayConstant.NULL;
+        }
+        return array;
+    }
+
+    @Override
+    public String getName() {
+        return "last";
+    }
+
+    @Override
+    public int getValueIndex() {
+        return valueIndex;
+    }
+
+    @Override
+    public void initValueIndex(int valueIndex) {
+        this.valueIndex = valueIndex;
+    }
+
+    @Override
+    public void initValueTypes(ArrayColumnTypes columnTypes) {
+        this.valueIndex = columnTypes.getColumnCount();
+        columnTypes.add(ColumnType.LONG);
+        columnTypes.add(ColumnType.LONG);
+    }
+
+    @Override
+    public boolean isConstant() {
+        return false;
+    }
+
+    @Override
+    public boolean isScalar() {
+        return false;
+    }
+
+    @Override
+    public boolean isThreadSafe() {
+        return false;
+    }
+
+    @Override
+    public void merge(MapValue destValue, MapValue srcValue) {
+        long srcRowId = srcValue.getLong(valueIndex);
+        long destRowId = destValue.getLong(valueIndex);
+        if (srcRowId != Numbers.LONG_NULL && (srcRowId > destRowId || destRowId == Numbers.LONG_NULL)) {
+            destValue.putLong(valueIndex, srcRowId);
+            destValue.putLong(valueIndex + 1, srcValue.getLong(valueIndex + 1));
+        }
+    }
+
+    public void setAllocator(GroupByAllocator allocator) {
+        sink.setAllocator(allocator);
+    }
+
+    @Override
+    public void setNull(MapValue mapValue) {
+        mapValue.putLong(valueIndex, Numbers.LONG_NULL);
+        mapValue.putLong(valueIndex + 1, 0);
+    }
+
+    public boolean supportsParallelism() {
+        return true;
+    }
+
+    @Override
+    public void toTop() {
```

Comment:
nit: this override is redundant since `UnaryFunction`'s implementation will be used anyway.